### PR TITLE
fabric:   IEvent, Transform, StaticCanvas

### PR DESCRIPTION
**Pull request by QJesus**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: < [toJSON is not a string, it is an object](https://github.com/fabricjs/fabric.js/blob/master/dist/fabric.js#L16153) >
- [x] add missing transform
- [x] ~~IEvent, e, it's type MouseEvent.            or  make on is generic, like canvas.on< MouseEvent >('object:modified', e => { /* e.e is MouseEvent */ });~~
image

### CODE REVIEW

File: types/fabric/fabric-impl.d.ts [hunk 1]
```diff
@@ -170,16 +170,18 @@ interface IDataURLOptions {
     withoutShadow?: boolean | undefined;
 }
 
-interface IEvent<E extends Event = Event> {
+interface IEvent<E extends Event = Event & MouseEvent> {
```
Comment (NotWoods) [inline types/fabric/fabric-impl.d.ts:173]: Why is this change needed?

**Review by NotWoods — CHANGES_REQUESTED** [review]
Please fill our the PR template and remove whitespace only changes

File: types/fabric/fabric-impl.d.ts (hunk 1 above)
Comment (QJesus) [reply types/fabric/fabric-impl.d.ts:173]: e.ctrlKey 
e.shiftKey 
e.altKey 
e.metaKey

Comment (QJesus) [reply types/fabric/fabric-impl.d.ts:173]: or make on is generic, like canvas.on< MouseEvent >('', e => { /* e.e is MouseEvent */ });

Comment (NotWoods) [reply types/fabric/fabric-impl.d.ts:173]: It's better to make on generic as you suggested, since not every event will be a mouse event. Thank you for the details!

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/fabric/tslint.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
         "no-empty-interface": false,
         "no-redundant-jsdoc-2": false,
         "no-unnecessary-class": false,
+        "no-unnecessary-generics": false,
```
[the remark this answers is not in the document]
Comment (QJesus) [reply types/fabric/tslint.json]: I don't know how to make it generic without this config. 
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63370#discussion_r1032239515
You can fix it

**Comment by andrewbranch** [comment]
Looks good to me. I’ll approve for self-merge this week if @NotWoods or another owner doesn’t object in the meantime.

**Review by andrewbranch — APPROVED** [review]

**Comment by QJesus** [comment]
> Ready to merge
